HARK/core.py

```diff
@@ -862,207 +1021,303 @@ def clear_history(self):
         for var_name in self.track_vars:
             self.history[var_name] = np.empty((self.T_sim, self.AgentCount)) + np.nan
 
+    clear_history_mcrlo = mcrlo_clear_history = clear_history
+
 
-def solve_agent(agent, verbose):
+def solve_agent(agent, messaging_level, quietly=False, **kwds):
     """
-    Solve the dynamic model for one agent type
-    using backwards induction.
-    This function iterates on "cycles"
-    of an agent's model either a given number of times
-    or until solution convergence
-    if an infinite horizon model is used
+    Solve the dynamic model for one agent type using backwards induction.
+
+    Iterates on "cycles" of an agent's model either a given number
+    of times or until solution convergence if an infinite horizon model is used
     (with agent.cycles = 0).
 
     Parameters
     ----------
     agent : AgentType
         The microeconomic AgentType whose dynamic problem
         is to be solved.
-    verbose : boolean
-        If True, solution progress is printed to screen (when cycles != 1).
+    messaging_level : logging level
+        Controls amount of output
+    quietly : boolean
+        If True, all printed output is suppressed.
 
     Returns
     -------
     solution : [Solution]
-        A list of solutions to the one period problems that the agent will
-        encounter in his "lifetime".
+        A list of solutions to one period/stage problems that the agent will
+        encounter in a "lifetime".
     """
     # Check to see whether this is an (in)finite horizon problem
     cycles_left = agent.cycles  # NOQA
     infinite_horizon = cycles_left == 0  # NOQA
-    # Initialize the solution, which includes the terminal solution if it's not a pseudo-terminal period
-    solution = []
-    if not agent.pseudo_terminal:
-        solution.insert(0, deepcopy(agent.solution_terminal))
+    # If we are resuming a solution process, a solution might already exist
+    if not hasattr(agent, 'solution'):  # Not resuming, create anew
+        # Initialize the solution, which includes the terminal solution
+        solution = []
+        pseudo = (agent.pseudo_terminal is True)
+        if not pseudo:  # Then it's a real solution; should be part of the list
+            solution.insert(0, deepcopy(agent.solution_terminal))
+        completed_cycles = 0  # NOQA
```

Comment:
This feature of being able to break off and restart a solution maybe can be split out and brought into HARK 1.0